OneSignalSDK/onesignal/in-app-messages/src/main/java/com/onesignal/inAppMessages/internal/backend/impl/InAppBackendService.kt

```diff
@@ -209,4 +198,75 @@ internal class InAppBackendService(
     ) {
         Logging.error("Encountered a $statusCode error while attempting in-app message $requestType request: $response")
     }
+
+    private suspend fun attemptFetchWithRetries(
+        baseUrl: String,
+        rywToken: String?,
+        sessionDurationProvider: () -> Long,
+    ): List<InAppMessage>? {
+        var attempts = 1
+        var retryLimit: Int? = null // Retry limit will be determined dynamically
```

Comment:
We really don't want a default? If so let's change this comment to say the retry limit is remote defined, via the IAM fetch response.

Reply:
@Koji23 suggested we simply don't retry at all if it's not provided remotely
https://github.com/OneSignal/OneSignal-Android-SDK/pull/2168#discussion_r1773848134

Reply:
I see, then `0` should be the default instead of `null`, as the default would be to not retry then.

There is also currently a bug in the code were if the backend doesn't return a value for retries it will result in an infinite loop. Using `0` then instead of `null` will fix this. There are others ways to prevent the infinite loop bug but setting this to `0` instead of `null` also prevent this mistake from happening in the future.

Reply:
Fixed